crates/orchestrator/src/spawner.rs

```diff
@@ -158,11 +162,40 @@ where
     println!("🚀 {} : metrics link {prometheus_uri}", node.name);
     println!("📓 logs cmd: tail -f {}/{}.log", base_dir, node.name);
     println!("\n");
+
+    let client = retry(|| async { OnlineClient::from_url(&ws_uri).await })
+        .await
+        .context(format!("Failed to connect to node rpc at {ws_uri}"))?;
+
+    let rpc = RpcClient::from_url(&ws_uri)
+        .await
+        .context(format!("Failed to connect to rpc client at {ws_uri}"))?;
+
     Ok(NetworkNode::new(
         node.name.clone(),
         ws_uri,
+        rpc,
+        client,
         prometheus_uri,
         node.clone(),
         running_node,
     ))
 }
+
+async fn retry<T, F, R, E>(connect: F) -> Result<R, E>
+where
+    T: Future<Output = Result<R, E>>,
+    F: Fn() -> T,
+{
+    let mut retries = 5;
+    loop {
+        match connect().await {
+            Err(_) if retries >= 0 => {
+                println!("Error connecting, retrying ...");
```

Comment:
should be added to the `crates/configuration/src/shared/constants.rs`

Reply:
Are you moving all &str into this file? This might make logging a bit more tedious, curious what you gain from this?